### ISSUE TITLE
Android P, not getting notifications

### ISSUE THREAD

**Issue by BloodyIron** [vector-im/riot-android#2512]
I'm writing this on behalf of a friend of mine, so testing will probably involve a bit of back and forth :P 

He's on a Pixel 2 and he just upgraded to Android P. He reports that now he's not getting notifications at all.

He reports that before he upgraded he generally was getting notifications.

Riot Android v0.8.12 (G-b146)

**Comment by bmarty** [comment]
Thx for reporting

**Comment by BloodyIron** [comment]
Hope it can get fixed! I use this with some friends to keep touch, and they're missing out ;D

**Comment by echto** [comment]
You're not alone. Notifications do not work on Android 8.1.0 either.

**Comment by echto** [comment]
Any news on this?

**Comment by BloodyIron** [comment]
I too would like to know.

**Comment by bmarty** [comment]
It's P1 for next sprint, starting this week

**Comment by bmarty** [comment]
Android P has a quite new notification management. Please ensure that Riot is allowed to display notifications (in the notification panel, click on `Manage notifications` at the bottom left).

Tested with Android P emulator, I correctly get the notification

**Comment by BloodyIron** [comment]
Well, does the default install of Riot not ask for, or get, sufficient notification permissions when installed? Or Android upgraded?

**Comment by bmarty** [comment]
AFAIK there is nothing to do to allow notification, but you may have blocked them ny mistake, it happens to me sometimes

**Comment by echto** [comment]
I want to clarify that LED notifications are not working, the icon notification does work but isn't much help when the screen is off, so yes I can understand why you think notification are good to go when using an emulator, but LED notifications are broken.

Also, here are some screenshots of what I see and experience.

https://imgur.com/a/wqs9eh9

What does "1 category deleted" mean?

And "Invalid State" appears when trying to select a different notification option in Riot.

**Comment by c-e-p-x-u-o** [comment]
Same bug. Android version 8.1.0. Lineages OS 15.1.
1. Led doesn't work. 
2. I see notification dot during startup process with permanent notification. But no notification dot for new messages.

**Comment by cpw** [comment]
I've noticed this over the last month or so as well. Since I received the 5.0.6 oxygenos update on my OP3, I'd say.

Riot appears to be able to function for a short time in the background, but once the phone enters a deeper sleep mode, I receive no notification until I reopen riot. All my notification dots are on high or medium (I have the "1 category deleted" text).

**Comment by BloodyIron** [comment]
Some of my friends are still having this issue too :/

**Comment by c-e-p-x-u-o** [comment]
Any update? It's really hard to use riot for 5 month since August!

**Comment by BloodyIron** [comment]
Are you referring to which sound file gets played, or something else? As in, we'll need to pick the sound byte that's played again, after the next update?

> @echto
> 
> > What does "1 category deleted" mean?
> 
> This was a consequence of changing sound of notification from Riot. It's not possible to change sound of a notification channel, so to do so, current implementation deletes the notification channel and recreates it with the new sound.
> 
> Note that this will be removed in the next version of Riot. You will have to change sound of the notification channel directly in the system notification setting of Riot application

**Comment by theLimitBreaker** [comment]
I have Nokia 6.1 with Android Pie. And Notifications are working

**Comment by c-e-p-x-u-o** [comment]
Led is fixed now (0.8.22 G-b162).
But the dot doesn't!

**Comment by echto** [comment]
https://github.com/vector-im/riot-android/releases/tag/v0.8.22

works!  

Nice touch with being able to backup keys to homeserver.

LED works. Notifications work. ADT :)

**Comment by c-e-p-x-u-o** [comment]
@echto does dot work?

**Comment by c-e-p-x-u-o** [comment]
What is "visual notification"? I'm asking about the dot on the icon.

**Comment by c-e-p-x-u-o** [comment]
It's shown only when riot is "synchronizing" and not for new messages.
![image](https://user-images.githubusercontent.com/14617699/52179360-a07eb600-27ea-11e9-9961-eb80380b18c8.png)

**Comment by echto** [comment]
I guessed notification because I didn't know what you meant by "dot".  I haven't a seen a dot now that you've specified what you are looking for.

And honestly I didn't know it was a feature. Lack of LED and audio notifications were the issues of concern with me.

**Comment by echto** [comment]
@c-e-p-x-u-o, I'm using Version 0.8.22 (G-). and I do not see a dot when Riot is synchronising.   Doesn't bother me though.

**Comment by bmarty** [comment]
Notification should work correctly on Android P (Riot 0.8.26), please reopen this issue if you observe any trouble

**Comment by benhubert** [comment]
Yesterday we received an update to Android 9 for our bq Aquaris X2 Pro, and it seems that the notifications don't work anymore since then.

Devices: bq Aquaris X2 Pro
Android Version: 9 (kernel 4.4.153-perf+, build 2.0.0_20190322-1025)
Riot Versions: 0.8.28 (F-47beeb658) and 0.8.29 (G-b172)

The issue is exactly as @cpw described:

> Riot appears to be able to function for a short time in the background, but once the phone enters a deeper sleep mode, I receive no notification until I reopen riot. All my notification dots are on high ~~or medium~~.

**Comment by gfelbing** [comment]
Same for me, notifications are broken since the Android P upgrade.
Message notifications do not work, only a stuck 'Synchronizing' notification is persistent.

Device BQ Aquaris X2 Pro
Android 9, kernel 4.4.153-perf+, build 2.0.0_20190322-1025
Riot 0.8.29 (G-b172)

**Comment by ethoms** [comment]
I am also experienceing notification iisue with Google play store version of Riot.im (0.9.3). It seems to effect Android 9 on my Nokia 6 (Android One, I think), but not Android 7.1.2 (LineageOS 14.1 hlte). Please see my comments on issue #3197 .

**Comment by ethoms** [comment]
I found some very limited improvement with the Play store version (0.9.3). I was previously disabling the Notification LED in phone settings (global) but Riot was set to use it. So now I have enabled LED notifications, I can get notifications when the app is running in task list, not focused and in sleep mode (screen off). But once it goes to a deeper sleep, no notifications at all. I have app set to "not optimised" in battery settings, and everything is set to run unrestricted anywhere I can find. Also, if I swipe the app closed to remove from task list, no notifications at all. It seems on Android 9 the Riot 0.9.3 gets killed off easily. But on Android 7.1.2 it does not.

I'll be sticking to F-Droid version0.8.26 for now on my Nokia 6. As for all the company users I'm about to rollout to, I think I'll have to sideload this APK from a link on our own website.

Notifications on Android really needs to get sorted out, on all versions of Android. If i wasn't so determined to get an open protocol working for us, I would just give up by now.
